bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -375,92 +381,97 @@ async fn extension_loop_active(
                         Event::Metric(event) => {
                             debug!("Metric event: {:?}", event);
                         }
-                        Event::Telemetry(event) => match event.record {
-                            TelemetryRecord::PlatformStart { request_id, .. } => {
-                                invocation_context_buffer.insert(InvocationContext {
-                                    request_id,
-                                    runtime_duration_ms: 0.0,
-                                });
-                            }
-                            TelemetryRecord::PlatformInitReport {
-                                initialization_type,
-                                phase,
-                                metrics,
-                            } => {
-                                debug!("Platform init report for initialization_type: {:?} with phase: {:?} and metrics: {:?}", initialization_type, phase, metrics);
-                                lambda_enhanced_metrics
-                                    .set_init_duration_metric(metrics.duration_ms);
-                            }
-                            TelemetryRecord::PlatformRuntimeDone {
-                                request_id,
-                                status,
-                                metrics,
-                                ..
-                            } => {
-                                if let Some(metrics) = metrics {
-                                    invocation_context_buffer
-                                        .add_runtime_duration(&request_id, metrics.duration_ms);
+                        Event::Telemetry(event) => {
+                            let event_time = event.time;
```

Comment:
maybe I can refactor this `event_time` to be passed directly, it's way too hard to see what changed here